MinecraftClient/Protocol/MojangAPI.cs

```diff
@@ -142,5 +143,91 @@ public static string UuidToCurrentName(string uuid)
 
             return tempDict;
         }
+
+        /// <summary>
+        /// Obtain links to skin, skinmodel and cape of a player.
+        /// </summary>
+        /// <param uuid="uuid">UUID of a player</param>
+        /// <returns>Dictionary with a link to the skin and cape of a player.</returns>
+        public static Dictionary<string, string> SkinInfo(string uuid)
+        {
+            Dictionary<string, string> tempDict = new Dictionary<string, string>();
+            Dictionary<string, Json.JSONData> textureDict;
+            string base64SkinInfo;
+            Json.JSONData decodedJsonSkinInfo;
+
+            // Perform web request
+            try
+            {
+                // Obtain the Base64 encoded skin information from the API. Discard the rest, since it can be obtained easier through other requests.
+                base64SkinInfo = Json.ParseJson(wc.DownloadString("https://sessionserver.mojang.com/session/minecraft/profile/" + uuid)).Properties["properties"].DataArray[0].Properties["value"].StringValue;
+            }
+            catch (Exception) { return tempDict; }
+
+            // Parse the decoded string to the JSON format.
+            decodedJsonSkinInfo = Json.ParseJson(System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(base64SkinInfo)));
+
+            // Assert temporary variable for readablity.
+            // Contains skin and cape information.
+            textureDict = decodedJsonSkinInfo.Properties["textures"].Properties;
+
+            // Can apparently be missing, if no custom skin is set.
+            // Probably for completely new accounts. 
+            // (Still exists after changing back to Steve or Alex skin.)
+            if (textureDict.ContainsKey("SKIN"))
+            {
+                // Add the URL leading to the texture of the ingame skin.
+                tempDict.Add("SkinURL", textureDict["SKIN"].Properties.ContainsKey("url") ? textureDict["SKIN"].Properties["url"].StringValue : string.Empty);
+
+                // Detect whether the playermodel is based on Steve or Alex.
+                // If the skin property contains metadata, which always contains "slim", it is an Alex based skin.
+                tempDict.Add("PlayerModel", textureDict["SKIN"].Properties.ContainsKey("metadata") ? "Alex" : "Steve");
+            }
+            // Tested it on several players, this case never occured.
+            else
+            {
+                // This player has assumingly never changed their skin.
+                // Probably a completely new account.
+                tempDict.Add("SkinURL", string.Empty);
+                tempDict.Add("PlayerModel", DefaultModelAlex(uuid) ? "Alex" : "Steve");
+            }
+
+            // If a cape exists, add it, otherwise leave string empty.
+            tempDict.Add("CapeURL",
+                textureDict.ContainsKey("CAPE") ? textureDict["CAPE"].Properties["url"].StringValue : string.Empty);
+
+            return tempDict;
```

Comment:
I think it would better not to return a string dictionary because it would be harder for the user (method caller) to understand what is inside the dictionary. One solution to this problem is to create a class for holding the result and return the class to the user.
For example,
```cs
class SkinInfo
{
    public readonly string SkinUrl;
    ...
}
```
and return it,
```cs
return new SkinInfo()
{
    SkinUrl = theUrl,
    ...
};
```
So that user can know what is inside by looking at the class definition.

The same thing for `GetMojangServiceStatus()` method above.